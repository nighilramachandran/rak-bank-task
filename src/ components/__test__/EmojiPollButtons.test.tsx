/* eslint-disable testing-library/no-node-access */
import { render, screen, fireEvent } from "@testing-library/react";
import EmojiPollButtons, { EmojiPollButtonsProps } from "../emoji-poll-buttons";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

// Create a mock store
const mockStore = configureStore([]);

const mockHandleEmotSelect = jest.fn();

const setup = (props: Partial<EmojiPollButtonsProps> = {}) => {
  const defaultProps: EmojiPollButtonsProps = {
    handleEmotSelect: mockHandleEmotSelect,
    _index: 0,
  };
  const mergedProps = { ...defaultProps, ...props };

  // Provide the mock store to the component
  const store = mockStore({
    Poll: {
      poll: [],
    },
  });

  render(
    <Provider store={store}>
      <EmojiPollButtons {...mergedProps} />
    </Provider>
  );
};

describe("EmojiPollButtons component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders buttons with icons and values", () => {
    setup();
    // expect(screen.getAllByTestId("btn-poll")).toHaveLength(3);
    expect(screen.getAllByTestId(/btn-poll-name/)).toHaveLength(3);
    expect(screen.getByText("Good")).toBeInTheDocument();
    expect(screen.getByText("Not Sure")).toBeInTheDocument();
    expect(screen.getByText("Bad")).toBeInTheDocument();
  });

  test("calls handleEmotSelect on button click", () => {
    setup();
    fireEvent.click(screen.getByText("Good"));
    expect(mockHandleEmotSelect).toHaveBeenCalledWith(
      { icon: "👍", value: "Good" },
      0
    );
  });

  test("changes opacity and transforms on icon hover", () => {
    setup();
    const textElement = screen.getByText("Good");
    const parentElement = textElement.parentElement;
    fireEvent.mouseOver(textElement);
    const computedStyles = window.getComputedStyle(textElement);
    const computedParentStyles = window.getComputedStyle(parentElement!);
    const opacity = computedStyles.getPropertyValue("opacity");
    const transform = computedParentStyles.getPropertyValue("transform");
    expect(opacity).toBe("1");
    expect(transform).toBe("translateY(-10px)");
  });

  test("resets opacity and transforms on icon mouse out", () => {
    setup();
    const textElement = screen.getByText("Good");
    fireEvent.mouseOver(textElement);
    fireEvent.mouseOut(textElement);
    const computedStyles = window.getComputedStyle(textElement);
    const opacity = computedStyles.getPropertyValue("opacity");
    expect(opacity).toBe("0");
  });
});
