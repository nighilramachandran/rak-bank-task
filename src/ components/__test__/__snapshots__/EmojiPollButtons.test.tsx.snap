// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiPollButtons component match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiStack-root css-7v40v5-MuiStack-root"
    >
      <div
        class="MuiBox-root css-c0z7sh"
        data-testid="btn-poll-name-Good-slide-0"
      >
        👍
        <p
          class="MuiTypography-root MuiTypography-body1 css-4t8l2l-MuiTypography-root"
        >
          Good
        </p>
      </div>
      <div
        class="MuiBox-root css-c0z7sh"
        data-testid="btn-poll-name-Not Sure-slide-0"
      >
        🤔
        <p
          class="MuiTypography-root MuiTypography-body1 css-4t8l2l-MuiTypography-root"
        >
          Not Sure
        </p>
      </div>
      <div
        class="MuiBox-root css-c0z7sh"
        data-testid="btn-poll-name-Bad-slide-0"
      >
        👎
        <p
          class="MuiTypography-root MuiTypography-body1 css-4t8l2l-MuiTypography-root"
        >
          Bad
        </p>
      </div>
    </div>
  </div>
</body>
`;
